ffi/athcon/bindings/rust/athcon-declare/src/lib.rs

```diff
@@ -336,16 +336,21 @@ fn build_execute_fn(name: &VMName) -> proc_macro2::TokenStream {
           use athcon_vm::AthconVm;
 
           // TODO: context is optional in case of the "precompiles" capability
-          if instance.is_null() || msg.is_null() || (code.is_null() && code_size != 0) {
+          if instance.is_null() || (code.is_null() && code_size != 0) {
               // These are irrecoverable errors that violate the athcon spec.
               std::process::abort();
           }
 
-          assert!(!instance.is_null());
-          assert!(!msg.is_null());
+          let host = if let Some(host) = unsafe { host.as_ref() } {
+            host
+          } else {
+            return athcon_vm::ExecutionResult::failure().into();
+          };
 
-          let execution_message: ::athcon_vm::ExecutionMessage = unsafe {
-              msg.as_ref().expect("athcon message is null").into()
+          let execution_message = if let Some(msg) = unsafe { msg.as_ref() } {
+            msg.into()
+          } else {
+            std::process::abort()
```

Comment:
Can we return a more meaningful error here, as before? Not sure if `abort()` takes an argument, but at the very least we could print to STDERR

Reply:
Changed the code to use assertions with an extra message, which is the recommended way for verifying runtime invariants in unsafe code: https://doc.rust-lang.org/std/macro.assert.html#uses